sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/streaming/StreamWriter.java

```diff
@@ -29,14 +31,15 @@
  * increasing numeric ID. This writer handles commits and aborts for each successive epoch.
  */
 @InterfaceStability.Evolving
-public interface StreamWriter extends DataSourceWriter {
```

Comment:
By design we would create more mix-in interfaces for `DataSourceWriter`, like `SupportsWriteColumnarBatch`. If the `StreamWriter` doesn't extend `DataSourceWriter`, how shall we apply those mix-in interfaces for `StreamWriter`?

Reply:
We should remove `extend DataSourceWriter` in these mix-in interfaces, so that it can work for both `DataSourceWriter` and `StreamWriter`.